lib/src/types/did_parser.rs

```diff
@@ -28,30 +53,94 @@ peg::parser! {
         ///   });
         /// ```
         pub rule ethr_did() -> Did
-            = method:method() ":" network:network()? account:account() { Did { method, network: network.unwrap_or_default(), account } }
+            = did() ":" method:method() ":" network:(chain_id() / network())? account:account() { Did { method, network: network.unwrap_or_default(), account } }
```

Comment:
In the old version, the network will be defaulted to `mainnet` when eluded. I assume we don't want to keep that logic now? I'm fine with that tho